src/commands/dev/headers.rs

```diff
@@ -20,15 +22,29 @@ pub fn strip_response_headers_prefix(parts: &mut ResponseParts) -> Result<(), fa
     Ok(())
 }
 
-pub fn prepend_request_headers_prefix(parts: &mut RequestParts) {
+pub fn prepend_request_headers_prefix(parts: &mut RequestParts, server_config: &ServerConfig) {
     let mut headers: HeaderMap = HeaderMap::new();
 
     for header in &parts.headers {
         let (name, value) = header;
+        let value = if name == "host" {
```

Comment:
while iterating over all of the headers and prefixing them with `cf-ew-raw-`, if it finds the `host` header, replace `localhost:8787` with whatever they passed in as the host. 

So if they ran `wrangler dev --host example.com --ip localhost --port 8787` and then ran `curl subdomain.localhost:8787`, it would find the incoming `host: subdomain.localhost:8787` and change it to `cf-ew-raw-host: subdomain.example.com`

This doesn't seem to behave as expected as `request.url` still seems to be `example.com` instead of `subdomain.example.com`.